test/framework/src/main/java/org/elasticsearch/test/TaskAssertions.java

```diff
@@ -70,6 +72,14 @@ public static void assertAllCancellableTasksAreCancelled(String actionPrefix) th
                 for (CancellableTask cancellableTask : taskManager.getCancellableTasks().values()) {
                     if (cancellableTask.getAction().startsWith(actionPrefix)) {
                         logger.trace("--> found task with prefix [{}]: [{}]", actionPrefix, cancellableTask);
+                        if (taskOpaqueIdMatches(cancellableTask.headers(), opaqueId) == false) {
```

Comment:
I would rather not leniently skip non-matching tasks: we shouldn't have any such tasks in the first place. If this is the problem then I'd prefer to track down where those tasks are coming from and get rid of them at source.

Let's assert that the task ID matches here instead.